データベース設計

ken-forest opened the pull request:
# What
ER図の作成と、README.mdファイルに
これから作るテーブルを記述

# Why
テーブル情報、作るべきテーブルのカラム、
モデルのファイルに記述するアソシエーション
などの情報を整理するため

＜Gyazoで撮影したER図＞
https://gyazo.com/68f7f3be6777fc73cc63be02bb53d574

ken-forest committed:
Draw.ioジェムを使ったER図の作成

ken-forest committed:
README.mdファイルにテーブル・アソシエーションを追加

ken-forest committed:
ER図、usersテーブルのカラム名のpassword→encrypted_passwordに変更

minachanchan commented on README.md:
```diff
@@ -5,20 +5,84 @@ application up and running.
+| --------------- | ------ | ------------------------- |
+| nickname        | string | null: false               |
+| email           | string | null: false, unique: true |
+| password        | string | null: false               |
```
```ruby
| password | string | null: false |
```

こちらのカラム名はdeviseで用意されている`encrypted_password`を使用しましょう！

minachanchan commented on README.md:
```diff
@@ -5,20 +5,84 @@ application up and running.
+| first_name      | string | null: false               |
+| last_name_kana  | string | null: false               |
+| first_name_kana | string | null: false               |
+
```
こちらの7つの項目に関してカラムの指定ができていていいですね！
現在後1つカラムが不足しております。
見本アプリを参照すると、現在指定頂いた項目の他にも入力欄があることがわかります。
不足しているものを確認し、改めて必要なカラムを考えてみましょう！

minachanchan commented on README.md:
```diff
@@ -5,20 +5,84 @@ application up and running.
+| ----------- | ---------- | ------------------------------ |
+| item_name   | string     | null: false                    |
+| detail      | text       | null: false                    |
+| price       | string     | null: false                    |
```
```ruby
| price | string | null: false |
```

商品価格は数値で扱うので、データ型は`integer`を指定しましょう！

minachanchan commented on README.md:
```diff
@@ -5,20 +5,84 @@ application up and running.
+| item_name   | string     | null: false                    |
+| detail      | text       | null: false                    |
+| price       | string     | null: false                    |
+| seller_name | string     | null: false                    |
```
```ruby
| seller_name | string | null: false |
```
出品者の名前をに入力する必要はございませんのでこちらのカラムは不要でございます。
削除しておきましょう！

minachanchan commented on README.md:
```diff
@@ -5,20 +5,84 @@ application up and running.
+| price       | string     | null: false                    |
+| seller_name | string     | null: false                    |
+| user        | references | null: false, foreign_key: true |
+
```
こちらのテーブルに関しても現在カラムが不足しております。
見本アプリのプルダウンで選択する項目について着目し、必要なカラムに関して改めて考えてみましょう！

minachanchan commented on README.md:
```diff
@@ -5,20 +5,84 @@ application up and running.
+
+| Column         | Type       | Options                        |
+| -------------- | ---------- | ------------------------------ |
+| buyer_name     | string     | null: false                    |
+| purchased_item | string     | null: false                    |
```
```ruby
| buyer_name     | string     | null: false                    |
| purchased_item | string     | null: false                    |
```
こちらのカラムは不要でございますので削除しておきましょう！
購入履歴に保存する「誰が」「どの商品」を購入した方という情報は、66、67行目のカラムに保存するデータが該当するためでございます。

minachanchan commented on README.md:
```diff
@@ -5,20 +5,84 @@ application up and running.
+| city            | string     | null: false                    |
+| address         | string     | null: false                    |
+| building        | string     |                                |
+| phone_number    | integer    | null: false                    |
```
```ruby
| phone_number | integer | null: false |
```
電話番号の多くは0から始まりますが、integer型は0から始まる整数を識別できません。
そのため、電話番号は`string`型で指定しましょう！

minachanchan commented on README.md:
```diff
@@ -5,20 +5,84 @@ application up and running.
+| building        | string     |                                |
+| phone_number    | integer    | null: false                    |
+| purchase_record | references | null: false, foreign_key: true |
+
```
こちらのテーブルも一点カラムが不足しております。
商品情報テーブルと同様にプルダウンで選択する項目について再度確認しましょう！

minachanchan reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。

8点コメントいたしましたので、ご確認頂けますと幸いでございます。

指摘内容の修正がお済みになりましたら、再度レビューのご依頼頂けますようお願い申し上げます。
その際、ER図もREADMEの内容と揃えていただき、次回のレビュー依頼の際に最新のER図のURLをまたこちらのプルリクエストに添付頂ければと存じます。

何卒よろしくお願いいたします。

ken-forest committed:
README.mdファイルのusersテーブルのpassword→encrypted_passwordに変更

ken-forest committed:
README.mdとfurima37120.dioのitemsテーブルからseller_nameのカラムを削除

ken-forest committed:
README.mdとfurima37120.dioのusersテーブルにbirth_dateカラムを追加

ken-forest committed:
README.mdのshipping_addressesテーブルのphone_numberカラムのtypeをinteger→stringに変更

ken-forest committed:
README.mdとfurima37120.dioのitemsテーブルにカラム５つ追加

＜追加したカラム＞
・item_category
・item_sales_status
・fee_status
・item_prefecture
・scheduled_delivery

ken-forest committed:
README.mdとfurima37120.dioのpurchase_recordsテーブルからカラム２つ削除

＜削除したカラム＞
・buyer_name
・purchased_item

ken-forest committed:
README.mdのitemsテーブルのpriceカラムのtypeをstring→integerに変更

ken-forest committed:
README.mdとfurima37120.dioのshipping_addressesテーブルにprefectureカラムを追加

ken-forest commented:
ご指摘いただいた、８つの項目を修正させていただきました。
引き続き、レビューをお願いします！

＜修正後のER図＞
https://gyazo.com/822b7d587fbba069c8505a4b294998d1

Saka-toku commented on README.md:
```diff
@@ -5,20 +5,88 @@ application up and running.
+| first_name         | string  | null: false               |
+| last_name_kana     | string  | null: false               |
+| first_name_kana    | string  | null: false               |
+| birth_date         | integer | null: false               |
```
誕生日の「年、月、日」はdate型を使用することで管理ができます。
カラムの型をdate型に変更しましょう！

Saka-toku commented on README.md:
```diff
@@ -5,20 +5,88 @@ application up and running.
+| ------------------ | ---------- | ------------------------------ |
+| item_name          | string     | null: false                    |
+| detail             | text       | null: false                    |
+| item_category      | string     | null: false                    |
+| item_sale_status   | string     | null: false                    |
+| fee_status         | string     | null: false                    |
+| item_prefecture    | string     | null: false                    |
+| scheduled_delivery | string     | null: false                    |
```
プルダウンで選択する箇所の値は「id」で扱われます。
ですのでカラム名を「単数形」＋「_id」に修正し、型を「数値型」に修正しましょう！

Saka-toku commented on README.md:
```diff
@@ -5,20 +5,88 @@ application up and running.
+| Column          | Type       | Options                        |
+| --------------- | ---------- | ------------------------------ |
+| postal_code     | string     | null: false                    |
+| prefecture      | string     | null: false                    |
```
`| prefecture | string | null: false |`

配送先のカラムは、itemsテーブルの発送元の地域のカラムと同じものを使用しましょう！
itemsテーブルのitem_prefectureカラムをprefecture_idに修正いただいても構いません。

【理由】
今回はアクティブハッシュを使用し都道府県情報をmodelファイルに用意します。
そしてmodelファイルを呼び出せば、複数の箇所で都道府県情報が使用できるようになるためです。

Saka-toku reviewed (commented):
レビュー依頼ありがとうございます！
修正点を確認させていただきました。
コメントいたしましたので、ご確認お願いいたします！

ken-forest committed:
README.mdとfurima37120.dioのshipping_addressesテーブルのカラム、prefecture→item_prefectureに変更

ken-forest committed:
README.mdとfurima37120.dioの修正

＜README.md＞
itemsテーブルとshippping_addressテーブルのprefectureを揃えた
＜furima37120.dio＞
itemsテーブルとshippping_addressテーブルのprefecture_idを揃えた

ken-forest committed:
README.mdのitemsテーブルとshippping_addressテーブルのprefectureに_idを足した

ken-forest committed:
README.mdとfurima37120.dioのitemsテーブルを修正

＜修正箇所＞
・プルダウンメニューに該当する５つのカラム名を全て、「単数系」+「id」に変更
・５つのカラムのtypeを全て、string→integerに変更

ken-forest committed:
README.mdのusersテーブルのbirth_dateカラムのtypeをinteger→dateに変更

ken-forest commented:
ご指摘いただいた３箇所を修正させてもらいました。
再々ですが、レビューをよろしくお願いします！

＜修正後のER図＞
https://gyazo.com/61edc7a409c077c6d1f2eaa1165373f8

CHURO-MAU commented on README.md:
```diff
@@ -5,20 +5,88 @@ application up and running.
+| Column          | Type       | Options                        |
+| --------------- | ---------- | ------------------------------ |
+| postal_code     | string     | null: false                    |
+| prefecture_id   | string     | null: false                    |
```
８３行目
こちらもアクティブハッシュを利用したカラムになりますので、Typeは整数型に変更しておきましょう！

CHURO-MAU reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。

ER図ですが、usersテーブルの上に不要なテーブルが重なっているようです。
ご確認いただき、usersテーブルが見える状態に修正をお願いいたします。

READMEに関しまして、細かいところですが
追加でコメントいたしましたので、ご確認お願いいたします。

ken-forest committed:
README.mdのshipping_addressesテーブルのprefecture_idカラムのtypeをstring→integerに変更

ken-forest committed:
ER図のテーブルを少し移動

ken-forest commented:
ご指摘いただいた１箇所、修正させていただきました。
引き続き、レビューをよろしくお願いします。
Gyazoの不備、申し訳ありませんでした。

＜修正後のER図＞
https://gyazo.com/df434dbec4052cc9161d15b56bb76537

AsukaToyoshima reviewed (commented):
レビューのご依頼ありがとうございます。
こちらLGTMとさせていただきます。お疲れ様でした！

次のブランチを切る前に、本プルリクエストのMergeとMasterブランチへのPullを忘れないようにしておきましょう。

引き続き、実装頑張っていきましょう！　応援しております！
